### PR TITLE
Minor code improvements

### PR DESCRIPTION
**Pull request by aaronryank**
- Magic numbers are bad so I added an enumerator in `interpreter.h` for clarity.
 - Having no blank lines in the source code is also bad so I added some of those too.

### CODE REVIEW

**Comment by Wheatwizard** [comment]
ec9adf4 seems helpful.
However I don't really see the point in ba3a4b6. There is very little on most lines so the addition of new blank lines seems to do nothing at best.

File: interpreter.c [hunk 1]
```diff
@@ -174,21 +173,24 @@ int interpreter_skip_loop(interpreter* interp, char first) {
...
 
 int interpreter_run(interpreter* interp) {
 	unsigned curly_depth = 0;
 	long last_index = -1;
 	char next, last = 0;
-	if (interp->status != 0) return interp->status;
-	while(!feof(interp->source)) {
+
+	if (interp->status != SUCCESS)
+            return interp->status;
```
Comment (1000000000) [inline interpreter.c]: This `if` statement should have brackets or it should be on one line.

File: interpreter.c [hunk 2]
```diff
@@ -306,16 +304,18 @@ int interpreter_run(interpreter* interp) {
...
 	data_stack_free(interp->data[1]);
+
 	if (interp->buf_size > 0) {
 		free(interp->buf);
 	}
-	free(interp);
-}
 
+	free(interp);
+}
```
Comment (1000000000) [inline interpreter.c]: There should be a newline at the end of a file.

File: interpreter.h [hunk 3]
```diff
@@ -2,9 +2,20 @@
 #define __interpreter_h_
 
 #include <stdio.h>
-
```
Comment (1000000000) [inline interpreter.h]: There should be a new line between the standard library includes and the others.

File: interpreter.h [hunk 4]
```diff
@@ -18,4 +29,4 @@ data_stack* interpreter_remove_active_stack(interpreter* interp);
 
 void interpreter_free(interpreter* interp);
 
-#endif//__interpreter_h_
+#endif //__interpreter_h_
```
Comment (1000000000) [inline interpreter.h]: The lack of space was so that it lined up with the include guard statements at the start of the program.

Comment (1000000000) [inline interpreter.h]: There should be a new line at the end of the file.

**Review by 1000000000 — CHANGES_REQUESTED** [review]
Looks pretty good, refactoring the error codes into an enum was definitely a good idea. There are just a few minor formatting things I'd like to see changed before we merge this.

**Comment by aaronryank** [comment]
@1000000000 Made the requested changes.

**Review by 1000000000 — APPROVED** [review]

**Pull request merged**